app/gui2/src/components/GraphEditor/GraphNodeMessage.vue

```diff
@@ -0,0 +1,44 @@
+<script setup lang="ts">
+/// types of messages
+export type GraphNodeMessageType = 'error' | 'warning'
+
+const props = defineProps<{ message: string; type: GraphNodeMessageType }>()
+
+function styleClassForType(type: GraphNodeMessageType) {
+  switch (type) {
+    case 'error':
+      return 'GraphNodeError'
+    case 'warning':
+      return 'GraphNodeWarning'
+    default:
+      return ''
+  }
+}
+</script>
+
+<template>
+  <div :class="'GraphNodeMessage ' + styleClassForType(props.type)" v-text="props.message"></div>
```

Comment:
```suggestion
  <div class="GraphNodeMessage" :class="styleClassForType(props.type)" v-text="props.message"></div>
```
String concatenation is so loosely typed in TS.